src/neo/Plugins/Plugin.cs

```diff
@@ -19,6 +20,7 @@ public abstract class Plugin : IDisposable
         internal static readonly List<IP2PPlugin> P2PPlugins = new List<IP2PPlugin>();
         internal static readonly List<IMemoryPoolTxObserverPlugin> TxObserverPlugins = new List<IMemoryPoolTxObserverPlugin>();
 
+        private static ImmutableList<object> services = ImmutableList<object>.Empty;
```

Comment:
```suggestion
        private static readonly ImmutableList<object> services = ImmutableList<object>.Empty;
```

Reply:
The `ImmutableList` can't be `readonly`.